### ISSUE TITLE
Upload Event Command Runner Not Triggered

### ISSUE THREAD

**Issue by cesboa** [filebrowser/filebrowser#2993]
**Description**
Some "Command runner" does not work. Like "After Upload" or "Before Upload", while "After Rename" and "After Delete Work"
Filebrowser 2.27.0 running via brew on latest macOS

**Expected behaviour**
All "Command runners" to work

**What is happening instead?**
Nothing happens

**How to reproduce?**
Add a command to eg. "After Upload" and upload something. The command is not executed.
See attached.
Only After Delete and After Rename makes a directory, the other two does not make a directory. stdout gives info on the ones who works, nothing in the other two.

![Screenshot 2024-02-07 at 14 20 32](https://github.com/filebrowser/filebrowser/assets/1929823/47e1c624-2ead-477a-ac23-9c3a4ae50e42)

**Comment by DaaGH** [comment]
Yes, I have the same thing. Docker-Alpine. v.2.24(s6)-2.27(s6). 2.23(s6) - ok.

**Comment by jantari** [comment]
I am having the same issue. My usecase is that all files should be (re)named to their sha256 hash automatically after upload, so I configured a "After Upload" command runner with the following:

```bash
mv -- "$FILE" "$(dirname -- "$FILE")/$(sha256sum -- "$FILE" | cut -d ' ' -f 1)"
```

this works when I run it inside the container via `docker exec -it filebrowser /bin/sh`, but when configured as a command runner there is nothing in the logs and the file isn't renamed, just nothing happens.

**Comment by jrukavina** [comment]
Same problem here, specifically for "After Upload" and "Before Upload" events.
Version 2.27.0

**Comment by brett-james-rocketlab** [comment]
Getting this problem as well. I got it to happen for before / after save, but never before / after upload.

**Comment by DgRosa** [comment]
I have the same problem, just trying to raise further awareness of this issue.

**Comment by RomanSonnik** [comment]
Is there any fixes for that?

**Comment by brett-james-rocketlab** [comment]
Yes, the problem came when chunked file support was added. So roll back to use I think it is 2.23 

.

Unfortunately brew doesn't have tsgged releases support for this repo, so I used docker instead and solved the problem.

**Comment by RussellLuo** [comment]
Just encountered the same issue! Based on the answer from @brett-james-rocketlab (thanks 🙏), I got it to work with the following command:

```bash
docker run -d --name=filebrowser \
    -v /path/to/root:/srv \
    -v /path/filebrowser.db:/database.db \
    -p 8080:80 \
    filebrowser/filebrowser:v2.23.0
```

**Comment by jantari** [comment]
Not stale

**Comment by ediazrod** [comment]
I have the same issue. no commands execute on after upload..

**Comment by brett-james-rocketlab** [comment]
This is still a problem.

**Comment by jantari** [comment]
Not stale, and won't be stale until it's fixed...

**Comment by maturbet** [comment]
The bug has been confirmed by more than 7 people and no changes have been made. How could it be stale ?

**Comment by Mubelotix** [comment]
Someone please DoS the bot

**Comment by rromrrom** [comment]
no commands run in my case too. Logs says nothing.

**Comment by kwisatz73** [comment]
Same here. Already thought I was too dumb to get it working. Happy to see that it seems to be a bug, unhappy to see that there is still no fix or workaround

**Comment by DuDu371** [comment]
Same problem, filebrowser 2.30 docker

**Comment by RoboticInterface** [comment]
I can Confirm this issue still persists on File Browser v2.31.2/129a4fd3 with the  "After Upload" and "Before Upload" events. I am using the latest docker version from https://hub.docker.com/r/filebrowser/filebrowser

**Comment by vjp23** [comment]
Commands are broken for me as well, but with various error messages.

**Comment by vjp23** [comment]
The error message I'm seeing in the logs is:

`2024/10/12 07:47:04 /testfile.txt: 404 [IP address] fork/exec /bin/sh -c: no such file or directory`

I was able to fix this by downgrading to as others have noted. The specific task I was trying to accomplish is detailed in https://github.com/filebrowser/filebrowser/issues/1854.

**Comment by parera10** [comment]
Commands for after_create don't work either. For after_copy and after_save they do.

**Comment by ERisEPR** [comment]
This is still an issue - downgrading did not work

**Comment by jussara-ti** [comment]
I have the same issue here. v2.31.2.

**Comment by jantari** [comment]
To everyone who keeps commenting that it's still an issue, please don't forget to :+1: the issue for visibility.

**Comment by jantari** [comment]
Not stale 🙃🙃🙃

**Comment by no6una9a** [comment]
Hello, same problem - no Save/Upload commands executed. Running latest filebrowser docker image behind nginx reverse proxy. Changing back to image v2.23.0 doesn't work either. Would be nice to have a feature fix for this wonderfull software. No6una9a

**Comment by jantari** [comment]
not stale

**Comment by i0ntempest** [comment]
Even with #3676 applied, filebrowser still reports "sh -c" being not found. Seem like word splitting issues.

**Comment by Mubelotix** [comment]
Nobody is working on this project anymore. If you want this fixed, you have to fix it yourself. If anybody comes up with a fix, please share it here though

**Comment by BeziCZ** [comment]
@i0ntempest Thank you for pointing it out. In the newest commit in my PR #3676 this works as well. Just added fix to that.

**Comment by i0ntempest** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/16017904?u=6d639b84a7f4975fdbd14ca3e2b2fc999ed29cbe&amp;v=4&amp;size=80">@i0ntempest](https://github.com/i0ntempest) Thank you for pointing it out. In the newest commit in my PR [#3676](https://github.com/filebrowser/filebrowser/pull/3676) this works as well. Just added fix to that.

Veryfied working - thanks!
I found another small bug - `after_create` command isn't run after new folder creations, only for new files. Could you tackle this as well?

**Comment by i0ntempest** [comment]
Not stale

**Comment by i0ntempest** [comment]
Not stale

**Comment by jantari** [comment]
Not stale

**Comment by Mubelotix** [comment]
Yeah it's stale because this project is abandoned

**Comment by hacdias** [comment]
Note: regression introduced by chunked file upload.
